src/compiler/parser.ts

```diff
@@ -1891,7 +1891,15 @@ namespace Parser {
                     }
 
                     if (pos >= 0) {
-                        const nonAwaitStatement = sourceFile.statements[pos];
+                        let nonAwaitStatement = sourceFile.statements[pos];
+                        // skip all statements in range already completely consumed by the reparsed statement
+                        for (let i = pos + 1; i < sourceFile.statements.length; i++) {
+                            if (nonAwaitStatement.end >= statement.end) {
+                                break;
+                            }
+                            pos = i;
+                            nonAwaitStatement = sourceFile.statements[pos];
+                        }

```

Comment:
I think this code is equivalent except that it runs the check one time before incrementing `pos`:
```suggestion
                        // skip all statements in range already completely consumed by the reparsed statement
                        while (sourceFile.statements[pos] < statement.end) {
                            pos++;
                        }
                        const nonAwaitStatement = sourceFile.statements[pos];
```